diff --git a/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt b/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
index d18b58a..b169992 100644
--- a/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
+++ b/node_modules/expo-av/android/src/main/java/expo/modules/av/ViewUtils.kt
@@ -8,7 +8,6 @@ import expo.modules.av.video.VideoView
 import expo.modules.av.video.VideoViewWrapper
 import expo.modules.core.ModuleRegistry
 import expo.modules.core.Promise
-import expo.modules.core.interfaces.services.UIManager
 
 object ViewUtils {
   interface VideoViewCallback {
@@ -18,7 +17,7 @@ object ViewUtils {
   @UiThread
   private fun tryRunWithVideoViewOnUiThread(moduleRegistry: ModuleRegistry, viewTag: Int, callback: VideoViewCallback, promise: Promise) {
     try {
-      val videoWrapperView = moduleRegistry.getModule(UIManager::class.java).resolveView(viewTag) as VideoViewWrapper?
+      val videoWrapperView = moduleRegistry.appContext?.findView<VideoViewWrapper>(viewTag)
       val videoView = videoWrapperView?.videoViewInstance
       if (videoView != null) {
         callback.runWithVideoView(videoView)
@@ -49,7 +48,7 @@ object ViewUtils {
   @UiThread
   private fun tryRunWithVideoViewOnUiThread(moduleRegistry: ModuleRegistry, viewTag: Int, callback: VideoViewCallback, promise: expo.modules.kotlin.Promise) {
     try {
-      val videoWrapperView = moduleRegistry.getModule(UIManager::class.java).resolveView(viewTag) as VideoViewWrapper?
+      val videoWrapperView = moduleRegistry.appContext?.findView<VideoViewWrapper>(viewTag)
       val videoView = videoWrapperView?.videoViewInstance
       if (videoView != null) {
         callback.runWithVideoView(videoView)
