diff --git a/node_modules/expo-task-manager/android/.gradle/9.2.0/checksums/checksums.lock b/node_modules/expo-task-manager/android/.gradle/9.2.0/checksums/checksums.lock
new file mode 100644
index 0000000..8c86d4b
Binary files /dev/null and b/node_modules/expo-task-manager/android/.gradle/9.2.0/checksums/checksums.lock differ
diff --git a/node_modules/expo-task-manager/android/.gradle/9.2.0/fileChanges/last-build.bin b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileChanges/last-build.bin differ
diff --git a/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.bin b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.bin
new file mode 100644
index 0000000..5f6f161
Binary files /dev/null and b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.bin differ
diff --git a/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.lock b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..ee8be64
Binary files /dev/null and b/node_modules/expo-task-manager/android/.gradle/9.2.0/fileHashes/fileHashes.lock differ
diff --git a/node_modules/expo-task-manager/android/.gradle/9.2.0/gc.properties b/node_modules/expo-task-manager/android/.gradle/9.2.0/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock b/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock
new file mode 100644
index 0000000..bd671c9
Binary files /dev/null and b/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/buildOutputCleanup.lock differ
diff --git a/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/cache.properties b/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/cache.properties
new file mode 100644
index 0000000..8a1714a
--- /dev/null
+++ b/node_modules/expo-task-manager/android/.gradle/buildOutputCleanup/cache.properties
@@ -0,0 +1,2 @@
+#Mon Sep 14 17:01:48 IST 2026
+gradle.version=9.2.0
diff --git a/node_modules/expo-task-manager/android/.gradle/vcs-1/gc.properties b/node_modules/expo-task-manager/android/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/repository/TasksPersistence.java b/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/repository/TasksPersistence.java
index a872533..3eb13d1 100644
--- a/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/repository/TasksPersistence.java
+++ b/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/repository/TasksPersistence.java
@@ -16,9 +16,10 @@ import expo.modules.interfaces.taskManager.TaskInterface;
 public class TasksPersistence {
 
   public void clearTaskPersistence(SharedPreferences preferences, String but) {
+    String safeBut = but != null ? but : "";
     Map<String, ?> map = preferences.getAll();
     for(String key: map.keySet()) {
-      if(!but.equals(key)) {
+      if(!safeBut.equals(key)) {
         preferences.edit().remove(key).apply();
       }
     }
